sim/lib/mobilitysim.py

```diff
@@ -580,6 +580,59 @@ def _find_contacts(self, mob_traces):
 
         return contacts
 
+    def find_contacts_of_indiv(self, mob_traces, indiv, tmin):
+        """Find contacts of possible infector 'indiv' after time 'tmin' in a given list `mob_traces` of `Visit`s.
+        Note that this function is called during simulation and requires the non-grouped mob_traces which in case
+        dynamic_tracing=True are stored in mob.all_mob_traces."""
+        # Group mobility traces by site
+        mob_traces_at_site = defaultdict(list)
+        indiv_trace_at_site = defaultdict(list)
+        visited_sites = []
+
+        for v in mob_traces:
+            if v.indiv == indiv:
+                indiv_trace_at_site[v.site].append(v)
+                if v.site not in visited_sites:
+                    visited_sites.append(v.site)
+            mob_traces_at_site[v.site].append(v)
+
+        contacts = InterLap()
+
+        # For each site s
+        for s in visited_sites:
+            if self.verbose:
+                print('Checking site '+str(s+1)+'/'+str(len(visited_sites)), end='\r')
+            if len(indiv_trace_at_site[s]) == 0:
+                continue
+
+            # Init the interval overlap matcher
+            inter = InterLap()
+            inter.update(mob_traces_at_site[s])
+            # Match contacts, as in _find_contacts vo is the possibly infected
+            for vo in indiv_trace_at_site[s]:
+                if vo.t_to_shifted > tmin:
+                    vo_time = (vo.t_from, vo.t_to_shifted)
+                    for v in list(inter.find(other=vo_time)):
+                        # Ignore contacts with same individual
+                        if v.indiv == vo.indiv:
+                            continue
+                        # Compute contact time
+                        c_t_from = max(vo.t_from, v.t_from)
+                        # c_t_to = min(v.t_to, vo.t_to_shifted)
+                        c_t_to = min(vo.t_to_shifted, v.t_to)
+                        if c_t_to > c_t_from and c_t_to > tmin:
+                            # Set contact tuple
+                            c = Contact(t_from=c_t_from,
+                                        t_to=c_t_to,
+                                        indiv_i=v.indiv,
+                                        indiv_j=vo.indiv,
+                                        id_tup=(v.id, vo.id),
+                                        site=s,
+                                        duration=c_t_to - c_t_from)
+                            # Add it to interlap
+                            contacts.update([c])
+        return contacts
+
```

Comment:
This function is very very similar / a copy of the main part of `_find_contacts`. For bug fixing and later changes it would be good to outsource this into one function and the call it both in `_find_contacts` and in `find_contacts_of_indiv`

Reply:
Agree with @larslorch.

Reply:
Good point! I'll change that